packages/web-app-files/src/components/SideBar/Shares/SpaceShares.vue

```diff
@@ -0,0 +1,135 @@
+    <template v-else>
+      <invite-collaborator-form v-if="currentUserCanShare" key="new-collaborator" class="oc-my-s" />
+      <template v-if="hasCollaborators">
+        <ul
+          id="files-collaborators-list"
+          class="oc-list oc-list-divider oc-overflow-hidden oc-m-rm"
+          :aria-label="$gettext('Share receivers')"
+        >
+          <li v-for="collaborator in currentFileOutgoingCollaborators" :key="collaborator.key">
+            <collaborator-list-item
+              :share="collaborator"
+              :modifiable="isModifiable(collaborator)"
+              @onDelete="$_ocCollaborators_deleteShare(collaborator)"
+            />
+          </li>
+        </ul>
+      </template>
+    </template>
+  </div>
+</template>
+
+<script>
+import { mapGetters, mapActions, mapState } from 'vuex'
+import { useStore } from 'web-pkg/src/composables'
+import { clientService } from 'web-pkg/src/services'
+import CollaboratorListItem from './Collaborators/ListItem.vue'
+import InviteCollaboratorForm from './InviteCollaborator/InviteCollaboratorForm.vue'
+import { spaceManager } from '../../../helpers/share'
+import { createLocationSpaces, isLocationSpacesActive } from '../../../router'
+import { useTask } from 'vue-concurrency'
+
+export default {
+  title: ($gettext) => {
+    return $gettext('Members')
+  },
+  name: 'SpaceShares',
+  components: {
+    CollaboratorListItem,
+    InviteCollaboratorForm
+  },
+  inject: ['displayedItem'],
+  setup() {
+    const store = useStore()
+    const graphClient = clientService.graphAuthenticated(
+      store.getters.configuration.server,
+      store.getters.getToken
+    )
+
+    const loadSharesTask = useTask(function* (signal, ref) {
+      yield ref.loadCurrentFileOutgoingShares({
+        client: graphClient,
+        path: ref.space.id,
+        space: ref.space
+      })
+    })
+
+    return { graphClient, loadSharesTask }
+  },
+  computed: {
+    ...mapGetters('Files', [
+      'highlightedFile',
+      'currentFileOutgoingCollaborators',
+      'currentFileOutgoingSharesLoading',
+      'sharesTreeLoading'
+    ]),
+    ...mapState(['user']),
+    space() {
+      return this.displayedItem.value
+    },
+    sharesLoading() {
+      return this.currentFileOutgoingSharesLoading
+    },
+    hasCollaborators() {
+      return this.currentFileOutgoingCollaborators.length > 0
+    },
+    currentUserCanShare() {
+      return this.currentUserIsManager
+    },
+    currentUserIsManager() {
+      const currentUserCollaborator = this.currentFileOutgoingCollaborators.find(
+        (collaborator) => collaborator.collaborator.name === this.user.id
+      )
+
+      return currentUserCollaborator?.role?.name === spaceManager.name
+    }
+  },
+  mounted() {
+    this.loadSharesTask.perform(this)
+  },
+  methods: {
+    ...mapActions('Files', ['loadCurrentFileOutgoingShares', 'deleteShare']),
+
+    isModifiable(share) {
+      if (this.currentFileOutgoingCollaborators.length < 2) {
```

Comment:
Nice idea for an early return, but I don't think that it's worth it. When the list only contains 1 item then filtering is fast anyway. I'd be fine with removing this early return in favour of less code complexity.